### PR TITLE
Don't predict memory in effective hpa

### PR DESCRIPTION
**Pull request by qmhu**

### CODE REVIEW

File: pkg/recommend/advisor/ehpa.go [hunk 1]
```diff
@@ -96,11 +96,11 @@ func (a *EHPAAdvisor) Advise(proposed *types.ProposedRecommendation) error {
 	targetCpuUtilization := int32(50) // todo: configurable
 	maxReplicasFactor := 1.2          // todo: configurable
 
-	maxCpuUtilization := int32(int64(maxCpuUsage) * 1000 * 100 / requestTotal)
+	maxCpuUtilization := int32(int64(maxCpuUsage) * 1000 * 100 * int64(len(a.Pods)) / requestTotal)
```
Comment (kitianFresh) [inline pkg/recommend/advisor/ehpa.go:99]: why multiply pods num. maxCpuUsage is average max usage of each pod?  I see it is computed only by one timeseries tsList[0], not all

File: pkg/recommend/advisor/ehpa.go [hunk 2]
```diff
@@ -96,11 +96,11 @@ func (a *EHPAAdvisor) Advise(proposed *types.ProposedRecommendation) error {
...
 	proposedMaxRatio := float64(maxCpuUtilization) / float64(targetCpuUtilization)
 	maxReplicasProposed := int32(math.Ceil(proposedMaxRatio * float64(a.ReadyPodNumber) * maxReplicasFactor))
 
-	minCpuUtilization := int32(int64(minCpuUsage) * 1000 * 100 / requestTotal)
+	minCpuUtilization := int32(int64(minCpuUsage) * 1000 * 100 * int64(len(a.Pods)) / requestTotal)
```
Comment (kitianFresh) [inline pkg/recommend/advisor/ehpa.go:103]: same fuzzing as above

File: pkg/controller/ehpa/hpa.go [hunk 3]
```diff
@@ -189,7 +189,7 @@ func (c *EffectiveHPAController) GetHPAMetrics(ctx context.Context, ehpa *autosc
 			if metric.Type == autoscalingv2.ResourceMetricSourceType {
 				name, err := GetPredictionMetricName(metric.Resource.Name)
 				if err != nil {
-					return nil, err
+					continue
```
Comment (mfanjie) [inline pkg/controller/ehpa/hpa.go:192]: log error, suggest not make it silent

File: pkg/controller/ehpa/predict.go [hunk 4]
```diff
@@ -131,7 +131,7 @@ func (c *EffectiveHPAController) NewPredictionObject(ehpa *autoscalingapi.Effect
 		if metric.Type == autoscalingv2.ResourceMetricSourceType {
 			metricName, err := GetPredictionMetricName(metric.Resource.Name)
 			if err != nil {
-				return nil, err
+				continue
```
Comment (mfanjie) [inline pkg/controller/ehpa/predict.go:134]: same as above

File: pkg/recommend/advisor/ehpa.go (hunk 1 above)
Comment (qmhu) [reply pkg/recommend/advisor/ehpa.go:99]: requestTotal contains total request for workload, so we need to add this.

Comment (kitianFresh) [reply pkg/recommend/advisor/ehpa.go:99]: maxCpuUsage is max of workload‘s(sum of all pods) usage too, so the original code is right, do not need fix

**Pull request merged**